src/terms/terms.jl

```diff
@@ -67,5 +50,63 @@ breaks_symmetries(term_builder::Magnetic) = true
 breaks_symmetries(term_builder::ExternalFromReal) = true
 breaks_symmetries(term_builder::ExternalFromFourier) = true
 
-### Builders are objects X that store the term parameters, and produce a
-# XTerm <: Term when instantiated with a `basis`
+
+
+
+# forces computes either nothing or an array forces[el][at][α]
+function forces(term::Term, ψ, occ; kwargs...)
+    nothing  # by default, no force
+end
+@timing function forces(basis::PlaneWaveBasis, ψ, occ; kwargs...)
+    if !any(iszero(kpt.coordinate) for kpt in basis.kpoints)
+        @warn "Forces for shifted k-Grids not tested"
+    end
+
+    # TODO optimize allocs here
+    T = eltype(basis)
+    f = [zeros(Vec3{T}, length(positions)) for (type, positions) in basis.model.atoms]
+    for term in basis.terms
+        ft = forces(term, ψ, occ; kwargs...)
+        if ft !== nothing
+            f += ft
+        end
+    end
+    f
+end
+forces(scfres) = forces(scfres.ham.basis, scfres.ψ, scfres.occupation, ρ=scfres.ρ)
+
+
+
+
+"""
+    compute_kernel(term::Term; kwargs...)
+
+Computes a matrix representation of the full response kernel
+(derivative of potential with respect to density) in real space.
+"""
+function compute_kernel(term::Term; kwargs...)
+    false  # "strong zero": false + something = something
```

Comment:
What's the difference to using `0`

Reply:
Try `-0.0 + 0` vs `-0.0 + false`. false also allows more optimizations.

Reply:
Is this documented somewhere?

Reply:
It's part IEEE-whatever semantics, part julia promotion rules. Not sure of any explicit docs

Reply:
ok. Thanks ... good to know nevertheless.